infra/azure-peerpods/main.tf

```diff
@@ -51,23 +51,35 @@ resource "azuread_application" "app" {
 }
 
 resource "azuread_service_principal" "sp" {
-  application_id               = azuread_application.app.application_id
+  client_id                    = azuread_application.app.client_id
   app_role_assignment_required = false
   owners                       = [data.azuread_client_config.current.object_id]
 }
 
-resource "azurerm_role_assignment" "ra" {
+resource "azurerm_role_assignment" "ra_vm_contributor" {
   scope                = azurerm_resource_group.rg.id
-  role_definition_name = "Contributor"
+  role_definition_name = "Virtual Machine Contributor"
+  principal_id         = azuread_service_principal.sp.id
+}
+
+resource "azurerm_role_assignment" "ra_reader" {
+  scope                = azurerm_resource_group.rg.id
+  role_definition_name = "Reader"
+  principal_id         = azuread_service_principal.sp.id
+}
+
+resource "azurerm_role_assignment" "ra_network_contributor" {
+  scope                = azurerm_resource_group.rg.id
+  role_definition_name = "Network Contributor"
   principal_id         = azuread_service_principal.sp.id
 }
 
 resource "azuread_application_federated_identity_credential" "federated_credentials" {
   display_name   = local.name
-  application_id = azuread_application.app.object_id
-  audiences      = ["api://AzureADTokenExchange"]
+  application_id = azuread_application.app.id
   issuer         = azurerm_kubernetes_cluster.cluster.oidc_issuer_url
   subject        = "system:serviceaccount:confidential-containers-system:cloud-api-adaptor"
+  audiences      = ["api://AzureADTokenExchange"]
 }
 
 resource "azurerm_role_assignment" "ra_image" {
```

Comment:
I believe `ra_image` is not sufficient for read permissions on the image - at least I saw corresponding failures while testing that went away after specifying the image here directly. We can sort this out later, though.